### DISCUSSION TITLE
How to invoke the effect of `pytest --cache-clear` for a single fixture.

### DISCUSSION THREAD

**Discussion by alexander-soare** [pytest-dev/pytest#10636] [Q&A] [answered]
I read about `pytest --cache-clear` from the [documentation](https://docs.pytest.org/en/7.1.x/how-to/cache.html#cache) `pytest --cache-clear`. But I want to know if there's a way to specify that the return value of a specific fixture should not be cached in between test runs. It doesn't look like [fixture scopes](https://docs.pytest.org/en/6.2.x/fixture.html#fixture-scopes) are the answer (if I'm interpreting this documentation correctly).

**Comment by RonnyPfannschmidt** [comment, marked as answer]
It's not clear what you are asking for, the cache is on disk, fixtures are not

A fixture which uses caches would need a dedicated cleanup command as well

**Reply by alexander-soare** [reply]
Yeah so maybe I don't full understand what's happening. As I understand, a fixture will cache the return value in memory within a test run. But it seems there is another caching mechanism which caches to disk so that fixture return values may be used between test runs right? And the way to clear the latter cache is to use `--cache-clear`. If everything is correct so far, my question is how I can prevent the second caching mechanism from happening with a specific fixture (while keeping the first caching mechanism).

**Reply by RonnyPfannschmidt** [reply]
There is no built-in on dis cache for fixtures

It's technically not possible to do it in a safe /general way

**Reply by alexander-soare** [reply]
Doesn't [this](https://docs.pytest.org/en/7.1.x/how-to/cache.html#the-new-config-cache-object) mechanism need a disk cache?

**Reply by RonnyPfannschmidt** [reply]
This is a explicit usage of the cache in the fixture itself, i mentioned those before, they need a custom cleanup command if you want to do pinpointed cleanup

The fixture system does not know about the cache system.

**Reply by alexander-soare** [reply]
Ah I'm with you now. Thank you.
